### PR TITLE
Shared random number generator

### PR DESCRIPTION
**Pull request by tsengj10**
There is now a shared random number generator in the Node class.  Plugins should use `Node.rng` for all random number generation.  The seed is optionally specified on the command line.

I also pythonized the name True_VS_Fit to TrueVsFit.

I've removed the seed argument from the configuration files.  test-gen-config, test-liq-config, and test-csv.csv work, though test-csv1.csv doesn't.  @mcolomermolla could you please fix TrueVsFit so it is more generic, and it uses the right key in payload (perhaps you updated this to use 'tdelay' in your own branch?)?  Also one should use the logger instead of print.

### CODE REVIEW

**Comment by tsengj10** [comment]
Also found that when I changed to incrementing burst_id in #39, this caused the DAG to be recreated afresh for each trial. Which sounded like a good idea, except that it also meant that it was impossible to accumulate any histograms. So I've changed burst_id in Simple and SimpleTrials to always be 0, and incremented a trial_id instead. This trial_id is used in creating the filename in the renderers for Histogram1D and TimeProfile.

**Comment by mcolomermolla** [comment]
This is strange, TrueVSFit was using same right keys as previous steps of the chain, not my own. 
Actually I think that module was verified before it was merged to the main.
I will have a look at it and see what is the problem in csv1. 
What do you mean by more generic? I think it is generic enough, it just inputs:
- the true time delay (generated randomly in GenerateSGBG)
- the fit time delay from TimeDiff
and outputs the difference between them.

I will commit new changes into this same branch then, instead of starting from master.

**Comment by mcolomermolla** [comment]
> This is strange, TrueVSFit was using same right keys as previous steps of the chain, not my own.
> Actually I think that module was verified before it was merged to the main.
> I will have a look at it and see what is the problem in csv1.
> What do you mean by more generic? I think it is generic enough, it just inputs:
> 
> * the true time delay (generated randomly in GenerateSGBG)
> * the fit time delay from TimeDiff
>   and outputs the difference between them.
> 
> I will commit new changes into this same branch then, instead of starting from master.

Actually, you can see that ‘in_field’:’tdelay’ is also used in test-csv.csv, so the key is not the issue I would say...

**Comment by mcolomermolla** [comment]
What I spotted as differences: 

- TimeSeries (used in test-csv), uses:

`tdelay = data['sig_t_delay']
ngen = { 'times': a, 'gen_t_delay': tdelay }`

- GenerateSGBG uses:

`    self.tdelay = int(Node.rng.uniform(-20.0, 20.0))
    t_true = self.tdelay
    ngen = { 'times': a, 't_true': t_true }`

- As tdelay is used for the output of the tdelay estimate, and not for the true one, here I named it t_true. I think we should keep that name
- What I don't understand is why test-csv is working and not test-csv1 because in Combine.py, what is read y t_true, and not the key used in TimeSeries:

`      if 't_true' in d:
        data['t_true'] = d['t_true']
`

**Comment by mcolomermolla** [comment]
TrueVSFit has been corrected and now tes-csv1 works

Consistency between TimeSeries and GenerateSGBG generators concerning key names missing.

File: snewpdag/plugins/TrueVsFit.py [hunk 1]
```diff
@@ -56,15 +56,11 @@ def update(self, data):
     for n,i in enumerate(self.map):
        if(i=='JUNO'):
```
Comment (tsengj10) [inline snewpdag/plugins/TrueVsFit.py:57]: This was the line which made me think about how generic the code was.  Do we need to tie these lines to a particular experiment?

**Comment by tsengj10** [comment]
Thanks for fixing test-csv1!  Now works for me as well.

The name of the true time field was quite as important in test-csv, I think, because it doesn't actually use the true value for comparison, whereas test-csv1 does.

I was thinking a bit more about the field names again and how to lay out the simulation part of the DAG (thoughts in wiki are still in working draft stage). Will make a suggestion a bit later.

File: snewpdag/plugins/TrueVsFit.py (hunk 1 above)
Comment (mcolomermolla) [reply snewpdag/plugins/TrueVsFit.py:57]: okey, now I see... 
indeed I did not find an optimal way of implementing this as there are different input fields to define dt1, dt2 and dt_fit.
I have implemented it in a way which makes no reference to detector names, let me know what you think.

**Comment by mcolomermolla** [comment]
> Thanks for fixing test-csv1! Now works for me as well.
> 
> The name of the true time field was quite as important in test-csv, I think, because it doesn't actually use the true value for comparison, whereas test-csv1 does.
> 
> I was thinking a bit more about the field names again and how to lay out the simulation part of the DAG (thoughts in wiki are still in working draft stage). Will make a suggestion a bit later.

No problem, I have also made it more general following your comment. 

Yes - let's have a proposal for the field names! 

Can you give a bit more details about how the random seed works now from the DAG?

**Comment by tsengj10** [comment]
I'll try to write a bit more later, but the basic idea is to use Node.rng as the random number generator, rather than self.rng.  One then sets the random number seed from the command line (--seed 1234) rather than in the configuration.

**Pull request merged**